packages/graph/src/Graph.ts

```diff
@@ -856,7 +857,24 @@ export class Graph extends SVGZoomWidget {
     }
 
     //  Events  ---
-    graph_selection(_selection) {
+    centroids(): Vertex[] {
+        return this._graphData.vertices().filter(vertex => vertex.centroid());
+    }
+
+    selectionChanged() {
+        if (this.highlightSelectedPathToCentroid()) {
+            const highlightedEdges = {};
+            this.centroids().forEach(centroid => {
+                this.selection().forEach(selection => {
+                    this._graphData.undirectedShortestPath(centroid.id(), selection.id()).forEach(e => {
+                        highlightedEdges[e.id()] = true;
```

Comment:
I don't see the logic to deselect the previously highlighted?

Reply:
Its on line 875 - every edge is visited and the class ".graphEdge" is either added or removed.